SL/RP.pm

```diff
@@ -1074,8 +1074,11 @@ sub get_accounts {
   # remove accounts with zero balance
   foreach $category (@{ $categories }) {
     foreach $accno (keys %{ $form->{$category} }) {
-      $form->{$category}{$accno}{last} = $form->round_amount($form->{$category}{$accno}{last}, $form->{decimalplaces});
-      $form->{$category}{$accno}{this} = $form->round_amount($form->{$category}{$accno}{this}, $form->{decimalplaces});
+      # armaghan 17 aug 2020 forced 8 precision in rounding to avoid rounding errors in calculating income
+      #$form->{$category}{$accno}{last} = $form->round_amount($form->{$category}{$accno}{last}, $form->{decimalplaces});
+      #$form->{$category}{$accno}{this} = $form->round_amount($form->{$category}{$accno}{this}, $form->{decimalplaces});
+      $form->{$category}{$accno}{last} = $form->round_amount($form->{$category}{$accno}{last}, 8);
```

Comment:
Hey Armaghan,

I propose to change it to 
if $form->{decimalplaces} < 8 {
      $form->{$category}{$accno}{last} = $form->round_amount($form->{$category}{$accno}{last}, 8);
     $form->{$category}{$accno}{this} = $form->round_amount($form->{$category}{$accno}{this}, 8);
} else {
$form->{$category}{$accno}{last} = $form->round_amount($form->{$category}{$accno}{last}, $form->{decimalplaces});
$form->{$category}{$accno}{this} = $form->round_amount($form->{$category}{$accno}{this}, $form->{decimalplaces});
}


So in case someone uses 20 decimals (why ever he would do so) it wouldn't break anything.

Reply:
I personally don't think we will ever need this check. And if we ever need it, we can adjust the code.